### PR TITLE
gcc-11.2.0: build bootloader libraries

### PR DESCRIPTION
**Pull request by davidkaroly**
beware: I haven't checked it on x86 secondary arch yet

### CODE REVIEW

**Comment by davidkaroly** [comment]
Successful build on x86_64 and x86 secondary arch - therefore promoting this pull request from Draft to Ready for Review.

**Comment by korli** [comment]
Looks like it triggered errors somehow: https://build.haiku-os.org/buildmaster/master/x86_64/logviewer.html?buildruns/6363/builds/76323.log

File: sys-devel/gcc/gcc-11.2.0_2021_07_28.recipe [hunk 1]
```diff
@@ -261,13 +266,57 @@ BUILD()
...
+	# deactivate threads and tls support
+	cp "../include/$effectiveTargetMachineTriple/bits/gthr-single.h" \
+		"../include/$effectiveTargetMachineTriple/bits/gthr-default.h"
+	sed -i -e 's,#define _GLIBCXX_HAS_GTHREADS 1,/* #undef _GLIBCXX_HAS_GTHREADS */,' \
+		"../config.h"
+	sed -i -e 's,#define _GLIBCXX_HAVE_TLS 1,/* #undef _GLIBCXX_HAVE_TLS */,' \
+		"../include/$effectiveTargetMachineTriple/bits/c++config.h"
+	make $jobArgs CFLAGS="-O2 $bootCcFlags" CXXFLAGS="-O2 $bootCcFlags"
+	mv .libs/libsupc++.a .libs/libsupc++-boot.a
+	mv saved/* .libs/
```
Comment (korli) [inline sys-devel/gcc/gcc-11.2.0_2021_07_28.recipe:318]: this is incorrect, please see how it is done for kernel.

Comment (davidkaroly) [reply sys-devel/gcc/gcc-11.2.0_2021_07_28.recipe:318]: Yes, I can see. We have to be more careful when "restoring" the original files after libsupc++-boot.a has been built.
I'll try to rework - and surely I'll do more in-depth testing before submitting it for review.

**Pull request merged**